v0/destinations/woopra/transform.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-param-reassign */
+const { set, get } = require("lodash");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  getErrorRespEvents,
+  getIntegrationsObj,
+  getSuccessRespEvents,
+  defaultGetRequestConfig
+} = require("../../util");
+
+const { EventType } = require("../../../constants");
+const { BASE_URL, mappingConfig, ConfigCategories } = require("./config");
+const { refinePayload } = require("./utils");
+
+const responseBuilder = async (payload, endpoint, method, projectName) => {
+  if (!payload) {
+    throw new CustomError("[ WOOPRA ]:: Parameters could not be found", 400);
+  }
+  if (!get(payload, "project")) {
+    set(payload, "Project", projectName);
+  }
+  const response = defaultRequestConfig();
+  response.params = removeUndefinedAndNullValues(payload);
+  response.endpoint = endpoint;
+  response.method = method;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/identify`;
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  refinePayload(message, payload);
+  projectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
+
+  return responseBuilder(
+    payload,
+    endpoint,
+    defaultGetRequestConfig.requestMethod,
+    projectName
+  );
+};
+const trackResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/ce`;
+  if (!message.event) {
+    throw new CustomError("[ WOOPRA ]:: Event Name can not be empty.", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.TRACK.name]
+  );
+  refinePayload(message, payload);
+  projectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
+  return responseBuilder(
+    payload,
+    endpoint,
+    defaultGetRequestConfig.requestMethod,
+    projectName
+  );
+};
+const pageResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/ce`;
+  if (!message.name) {
+    throw new CustomError("[ WOOPRA ]:: Page Name can not be empty.", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.PAGE.name]
+  );
+  refinePayload(message, payload);
+  projectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
```

Comment:
Don't modify config properties, create new variable

Reply:
Have changed the cost name to extractedprojectname